nexus/src/app/background/sync_switch_port_settings.rs

```diff
@@ -0,0 +1,1236 @@
+                            infra_ip_first: Ipv4Addr::UNSPECIFIED,
+                            infra_ip_last: Ipv4Addr::UNSPECIFIED,
+                            ports,
+                            bgp,
+                        }),
+                    },
+                };
+
+                // check to see if our config is different(?)
+                // we currently are not caching the last sent bootstore config, so we have to build
+                // it every time and compare.
+
+                let bootstore_needs_update = {
+                    match downstream_config {
+                        Some(ref existing_config) => {
+                            existing_config.schema_version != upstream_config.schema_version ||
+                            existing_config.body.ntp_servers != upstream_config.body.ntp_servers ||
+                            existing_config.body.rack_network_config != upstream_config.body.rack_network_config
+                        },
+                        _ => true,
+                    }
+                };
+
+                if bootstore_needs_update {
+                    let generation = match self.datastore
+                        .bump_bootstore_generation(opctx, NETWORK_KEY.into())
+                        .await {
+                        Ok(value) => value,
+                            Err(e) => {
+                                error!(
+                                    log,
+                                    "error while fetching next bootstore generation from db";
+                                    "key" => %NETWORK_KEY,
+                                    "error" => %e,
+                                );
+                                continue;
+                            },
+                        };
+
+                    upstream_config.generation = generation as u64;
+                    info!(
+                        &log,
+                        "updating bootstore config";
+                        "old config" => ?downstream_config,
+                        "new config" => ?upstream_config,
+                    );
+
+                    // push the updates to both scrimlets
+                    // if both scrimlets are down, bootstore updates aren't happening anyway
+                    for (location, client) in &sled_agent_clients {
+                        if let Err(e) = client.write_network_bootstore_config(&upstream_config).await {
+                            error!(
+                                log,
+                                "error updating bootstore";
+                                "location" => %location,
+                                "config" => ?upstream_config,
+                                "error" => %e,
+                            )
+                        }
+                    }
+                }
+            }
+            json!({})
+        }
+        .boxed()
+    }
+}
+
+fn uplinks(
```

Comment:
Just a note that wee need to re-evaluate

- oxidecomputer/customer-support#95

with this new code path for uplink setup in place.